driver/pelican/types/thermSchedule.go

```diff
@@ -0,0 +1,252 @@
+			}
+			for _, day := range []string{"Sunday", "Saturday"} {
+				thermSchedule.DaySchedules[day] = *weekend
+			}
+			weekday, weekdayError := getScheduleByDay(1, epnum, sitename, nodename, cookie, pel.timezone)
+			if weekdayError != nil {
+				return nil, fmt.Errorf("Error retrieving schedule for thermostat %v on weekday (day 1): %v", nodename, weekdayError)
+			}
+			for _, day := range []string{"Monday", "Tuesday", "Wednesday", "Thursday", "Friday"} {
+				thermSchedule.DaySchedules[day] = *weekday
+			}
+		} else {
+			return nil, fmt.Errorf("Failed to recognize repeat type of thermostat %v's schedule: %v", nodename, repeatType)
+		}
+
+		schedules[thermostatID.Id] = thermSchedule
+	}
+	return schedules, nil
+}
+
+func getSettings(sitename, thermostatID string, cookie *http.Cookie) (*settingsWrapper, error) {
+	var requestURL bytes.Buffer
+	requestURL.WriteString(fmt.Sprintf("https://%s.officeclimatecontrol.net/ajaxThermostat.cgi?id=", sitename))
+	requestURL.WriteString(thermostatID)
+	requestURL.WriteString(":Thermostat&request=GetSchedule")
+
+	resp, _, errs := gorequest.New().Get(requestURL.String()).Type("form").AddCookie(cookie).End()
+	if errs != nil {
+		return nil, fmt.Errorf("Failed to retrieve schedule settings for thermostat %v: %v", thermostatID, errs)
+	}
+	var result settingsRequest
+	decoder := json.NewDecoder(resp.Body)
+	if decodeError := decoder.Decode(&result); decodeError != nil {
+		return nil, fmt.Errorf("Failed to decode schedule settings for thermostat %v: %v", thermostatID, decodeError)
+	}
+	return &result.Userdata, nil
+}
+
+func getScheduleByDay(dayOfWeek int, epnum float64, sitename, thermostatID string, cookie *http.Cookie, timezone *time.Location) (*ThermostatDaySchedule, error) {
+	// Construct Request URL for Thermostat Schedule by Day of Week
+	var requestURL bytes.Buffer
+	requestURL.WriteString(fmt.Sprintf("https://%s.officeclimatecontrol.net/thermDayEdit.cgi?section=json&nodename=", sitename))
+	requestURL.WriteString(thermostatID)
+	requestURL.WriteString("&epnum=")
+	requestURL.WriteString(fmt.Sprintf("%.0f", epnum))
+	requestURL.WriteString("&dayofweek=")
+	requestURL.WriteString(strconv.Itoa(dayOfWeek))
+
+	// Make Request, Decode into Response Struct
+	resp, _, errs := gorequest.New().Get(requestURL.String()).Type("form").AddCookie(cookie).End()
+	if errs != nil {
+		return nil, fmt.Errorf("Failed to retrieve schedule for thermostat %v on day of week %v: %v", thermostatID, dayOfWeek, errs)
+	}
+	var result scheduleRequest
+	decoder := json.NewDecoder(resp.Body)
+	if decodeError := decoder.Decode(&result); decodeError != nil {
+		return nil, fmt.Errorf("Failed to decode schedule for thermostat %v on day of week %v: %v", thermostatID, dayOfWeek, decodeError)
+	}
+
+	// Transfer Response Struct Data into return struct
+	var daySchedule ThermostatDaySchedule
+	for _, block := range result.ClientData.SetTimes {
+		var returnBlock ThermostatBlockSchedule
```

Comment:
It would be more idiomatic to write this as
```go
returnBlock := ThermostatBlockSchedule{
    CoolSetting: block.CoolSetting,
    HeatSetting: block.HeatSetting,
    System: block.System,
}
```

Reply:
Yes! I'll remember in the future to instantiate my structs using this notation, it's a lot more clear.